### PR TITLE
refactor: replace logrus with new logger

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)

### Description
This change replaces the old logger with the new log package. This change is necessary so that the `logger` package can be gradually replaced by the new `log` package, both of which can exist at the same time.

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3079)

### CODE REVIEW

**Review by janos — COMMENTED** [review]
The build is failing because windowsEventLogger also needs to be adjusted.

**Comment by mrekucci** [comment]
> The build is failing because windowsEventLogger also needs to be adjusted.

Fixed.

**Review by notanatol — APPROVED** [review]

**Review by janos — COMMENTED** [review]
This changeset contains code from other pr-s, already merged, that are not related to the logger. Could this PR be rebased so that it contains only intended changes? This probably means that the target bench needs to be up to date with the master branch.

**Comment by mrekucci** [comment]
> This changeset contains code from other pr-s, already merged, that are not related to the logger. Could this PR be rebased so that it contains only intended changes? This probably means that the target bench needs to be up to date with the master branch.

Done.

File: pkg/p2p/libp2p/connections_test.go [hunk 1]
```diff
@@ -963,40 +960,40 @@ func TestWithBlocklistStreams(t *testing.T) {
...
-	// wait for logs to be written to buffers
-	for t := time.Now().Add(10 * time.Second); time.Now().Before(t); time.Sleep(50 * time.Microsecond) {
-		if s1Logs.Len() > 0 && s2Logs.Len() > 0 {
-			break
-		}
-	}
-
-	testUserAgentLogLine(t, s1Logs, "(inbound)")
-	testUserAgentLogLine(t, s2Logs, "(outbound)")
+	// TODO: restore the test when the transition to the new `log` package is done.
```
Comment (janos) [inline pkg/p2p/libp2p/connections_test.go:963]: I am afraid that this part of the code will be forgotten if left commented. Also this functionality is very important for discovery of the user agent and bee versions by the swarm scan service. I would stick with the principle of not merging commented code even if the merge is not to the master branch.

Comment (janos) [inline pkg/p2p/libp2p/connections_test.go:963]: I am afraid that this part of the code will be forgotten if left commented. Also this functionality is very important for discovery of the user agent and bee versions by the swarm scan service. I would stick with the principle of not merging commented code even if the merge is not to the master branch.

Comment (mrekucci) [reply pkg/p2p/libp2p/connections_test.go]: Fixed.

File: pkg/accounting/accounting_test.go [hunk 2]
```diff
@@ -102,7 +101,7 @@ func TestMutex(t *testing.T) {
 
 // TestAccountingAddBalance does several accounting actions and verifies the balance after each steep
 func TestAccountingAddBalance(t *testing.T) {
-	logger := logging.New(io.Discard, 0)
```
Comment (istae) [inline pkg/accounting/accounting_test.go:105]: why the need to change this? the new logger should continue to work with `io.Discard`.

Comment (mrekucci) [reply pkg/accounting/accounting_test.go:105]: This is just a temporary change and will be reverted when the migration is completed.

File: pkg/p2p/libp2p/connections_test.go [hunk 3]
```diff
@@ -963,22 +963,21 @@ func TestWithBlocklistStreams(t *testing.T) {
...
-
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	// use concurrent-safe buffers as handlers are logging concurrently
-	s1Logs := new(buffer)
-	s2Logs := new(buffer)
+	// use concurrent-safe buffer as handlers are logging concurrently
+	logSink := new(buffer)
+	logger := logging.New(logSink, 5)
```
Comment (janos) [inline pkg/p2p/libp2p/connections_test.go:971]: It is important that two services do not share the logger in order to evaluate which one has the `(inbound)` and which one `(outbound)` log message.

Comment (mrekucci) [reply pkg/p2p/libp2p/connections_test.go]: Fixed.

**Review by istae — APPROVED** [review]

**Review by istae — APPROVED** [review]
thanks!

**Review by janos — APPROVED** [review]
👍

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**